lib/OpenQA/WebAPI/Controller/Test.pm

```diff
@@ -22,25 +22,16 @@ use constant DEPENDENCY_DEBUG_INFO => $ENV{OPENQA_DEPENDENCY_DEBUG_INFO};
 
 sub referer_check ($self) {
     return $self->reply->not_found if (!defined $self->param('testid'));
-    my $referer = $self->req->headers->header('Referer') // '';
-    return 1 unless $referer;
+    return undef unless my $referer = $self->req->headers->header('Referer') // '';
```

Comment:
This looks like it would always have returned 1. Maybe this is somehow causing the tests to fail?